packages/pyright-internal/src/languageService/hoverProvider.ts

```diff
@@ -142,6 +144,12 @@ export class HoverProvider {
             if (type !== undefined) {
                 this._tryAddPartsForTypedDictKey(format, sourceMapper, evaluator, node, type, results.parts);
             }
+        } else if (node.nodeType === ParseNodeType.ModuleName || node.nodeType === ParseNodeType.Module) {
```

Comment:
hmmm.. shouldn't be inside of node.nodeType === ParseNodeType.Name ?

import A.B

B is Name node under moduleName?

Reply:
actually, I would put this inside of _addDocumentationPart

if getType return nothing, then, check whether it is name node under moduleName, and do what you are doing here.

Reply:
Sorry I'm missing what you mean. `_addDocumentationPart` isn't called for this scenario. It is for the A.B scenario you specified above. 

Are you suggesting if the nodeType is ModuleName or Module, it should call _addDocumentationPart which would then call this code if it couldn't find a type?

Reply:
I can't repro this anymore so not sure why I had to add this case. I'll add the changes you suggested into _addDocumentationPart.